content/posts/github-workflow-guide-for-modules.md

```diff
@@ -0,0 +1,239 @@
+
+```yaml
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+  schedule:
+    - cron: "15 4 * * 0" # Every Sunday morning
+  workflow_dispatch:
+```
+
+The `on` key controls when a workflow gets triggered. The options we've used here mean:
+
+- run every time someone pushes to any branch in the repository (this is code you or your colaborators write),
+- someone opens a pull request (also when they push or force push more code to their PR's branch),
+- automatically as a cronjob,
+- or with `workflow_dispatch` when you click the button to run it against a branch of your choosing.
+
+The `schedule` makes sense if you want to rebuild something automatically, such as documentation, a blog or a static site, or if your tests rely on external websites that might go away. In most cases you probably do not want to use it. Dave Cross' website [Planet Perl](https://perl.theplanetarium.org/) heavily uses this feature to aggregate Perl blog posts from around the web several times a day.
+
+```yaml
+jobs:
+```
+
+Next we have the `jobs`. This workflow defines three different ones, as shown above in the diagram. The first one is for building the distribution. This is not so much about running the tests, but about building the module once, so that it can be used in subsequent tests on different platforms. That way we can save time on complicated build processes.
+
+```yaml
+jobs:
+  build-job:
+    name: Build distribution
+    runs-on: ubuntu-20.04
+    container:
+      image: perldocker/perl-tester:5.34
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Tests
+        env:
+          AUTHOR_TESTING: 1
+          AUTOMATED_TESTING: 1
+          EXTENDED_TESTING: 1
+          RELEASE_TESTING: 1
+        run: auto-build-and-test-dist
+      - uses: actions/upload-artifact@v3
+        with:
+          name: build_dir
+          path: build_dir
+        if: ${{ github.actor != 'nektos/act' }}
+```
+
+The build job in our example runs on a specific Ubuntu version, on a custom Docker container that is maintained by the Perl 5 Porters named [perldocker](https://github.com/Perl/docker-perl-tester). It runs a specific Perl version that you chose in the `image` (this one uses 5.34), as well as the OS you set in the `runs-on` key. The container comes with a lot of different testing modules, CPAN clients such as [cpanm](https://metacpan.org/dist/App-cpanminus/view/lib/App/cpanminus/fatscript.pm) and [cpm](https://metacpan.org/dist/App-cpm/view/script/cpm) as well as spelling dictionaries and other useful stuff. This allows you to run most Perl related builds and tests pretty much out of the box, and cuts down on run time for individual jobs.
```

Comment:
The custom docker containers aren't maintained by P5P. Maybe the base container is? Not quite sure how all that works, but that repo lives in the `Perl` org mostly because of the bus number.